### ISSUE TITLE
[BUG] Network configuration enters undefined state (nodeNetwork stuck on terminating)

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#2080]
**Describe the bug**
Under certain circumstances the `networkNode` resource for a specific node may become stuck in `terminating` causing the network configuration on the OS to be left in an undefined state and thus disrupting VM connectivity.

**To Reproduce**
N/A

**Expected behavior**
`networkNode` resource should not be stuck in `terminating`state

**Actual behavior**
`networkNode` resource got stuck in `terminating` state, waiting on the `harvester-vip-controller` finalizer

**Support bundle**
N/A (sensitive content)

**Environment:**
Harvester 1.0.0
PXE

**Additional context**

harvester-network-manager:

```
2022-03-18T06:46:44.908473640Z time="2022-03-18T06:46:44Z" level=error msg="error syncing 'harvester-04-vlan': handler harvester-vip-controller: the vlan network hasn't removed, requeuing"
2022-03-18T06:46:44.914123796Z I0318 06:46:44.914064       1 controller.go:103] update vip interface, unavailable NIC: harvester-mgmt
2022-03-18T06:46:45.297406402Z I0318 06:46:45.297227       1 controller.go:59] cluster network configuration vlan has been changed
2022-03-18T06:53:51.503782045Z I0318 06:53:51.503651       1 controller.go:82] networkNode harvester-04-vlan is removed
2022-03-18T06:53:51.503802266Z time="2022-03-18T06:53:51Z" level=error msg="error syncing 'harvester-04-vlan': handler harvester-vip-controller: the vlan network hasn't removed, requeuing"
2022-03-18T07:10:31.504152724Z I0318 07:10:31.504051       1 controller.go:82] networkNode harvester-04-vlan is removed
2022-03-18T07:10:31.504272996Z time="2022-03-18T07:10:31Z" level=error msg="error syncing 'harvester-04-vlan': handler harvester-vip-controller: the vlan network hasn't removed, requeuing"
2022-03-18T07:27:11.504623558Z I0318 07:27:11.504528       1 controller.go:82] networkNode harvester-04-vlan is removed
2022-03-18T07:27:11.504651206Z time="2022-03-18T07:27:11Z" level=error msg="error syncing 'harvester-04-vlan': handler harvester-vip-controller: the vlan network hasn't removed, requeuing"
2022-03-18T07:43:51.505053685Z I0318 07:43:51.504952       1 controller.go:82] networkNode harvester-04-vlan is removed
2022-03-18T07:43:51.505089475Z time="2022-03-18T07:43:51Z" level=error msg="error syncing 'harvester-04-vlan': handler harvester-vip-controller: the vlan network hasn't removed, requeuing"
```

[harvester-network-manager.log](https://github.com/harvester/harvester/files/8349777/harvester-network-manager.log)

harvester-network-controller (problematic node):

```
2022-03-18T05:20:07.801785224Z I0318 05:20:07.801504       1 controller.go:75] regular reset node network
2022-03-18T05:46:06.765970867Z I0318 05:46:06.765772       1 controller.go:69] nad configuration vlan-2002-jitsi has been changed: {"cniVersion":"0.3.1","name":"vlan-2002-jitsi","type":"bridge","bridge":"harvester-br0","promiscMode":true,"vlan":2002,"ipam":{}}
2022-03-18T05:46:06.765988564Z I0318 05:46:06.765811       1 controller.go:69] nad configuration vlan-2101-rancher has been changed: {"cniVersion":"0.3.1","name":"vlan-2101-rancher","type":"bridge","bridge":"harvester-br0","promiscMode":true,"vlan":2101,"ipam":{}}
2022-03-18T05:46:06.766958683Z I0318 05:46:06.766917       1 controller.go:85] ignore link not found error, details: slave of harvester-br0 not found
2022-03-18T05:46:06.766964194Z I0318 05:46:06.766934       1 controller.go:69] nad configuration vlan-3001-cluster-1 has been changed: {"cniVersion":"0.3.1","name":"vlan-3001-cluster-1","type":"bridge","bridge":"harvester-br0","promiscMode":true,"vlan":3001,"ipam":{}}
2022-03-18T05:46:06.766975414Z I0318 05:46:06.766945       1 controller.go:85] ignore link not found error, details: slave of harvester-br0 not found
2022-03-18T05:46:06.766978263Z I0318 05:46:06.766958       1 controller.go:69] nad configuration default-vlan has been changed: {"cniVersion":"0.3.1","name":"default-vlan","type":"bridge","bridge":"harvester-br0","promiscMode":true,"vlan":1,"ipam":{}}
2022-03-18T05:46:06.767408802Z I0318 05:46:06.767365       1 controller.go:85] ignore link not found error, details: slave of harvester-br0 not found
2022-03-18T05:46:06.767523324Z I0318 05:46:06.767496       1 controller.go:85] ignore link not found error, details: slave of harvester-br0 not found
2022-03-18T05:50:07.802217550Z I0318 05:50:07.801937       1 controller.go:75] regular reset node network
2022-03-18T06:20:07.801196038Z I0318 06:20:07.801029       1 controller.go:75] regular reset node network
2022-03-18T06:50:07.801724779Z I0318 06:50:07.801529       1 controller.go:75] regular reset node network
2022-03-18T07:20:07.801375306Z I0318 07:20:07.801185       1 controller.go:75] regular reset node network
2022-03-18T07:50:07.801765840Z I0318 07:50:07.801602       1 controller.go:75] regular reset node network
```

**Comment by yaocw2020** [comment]
Once the nodenetwork is deleted, the harvester-nodenetwork-controller will tear down the VLAN and set the status as false. Then, harvester-vip-controller confirms the status and switch the VIP interface. In this case, the status seems not to be updated and the harvester-vip-controller returned the error. That’s the reason why it’s stuck in the terminal state. However, I could not figure out what happen at the time of deleting the nodenetwork from the provided log.
By the way, we realized the harvester-vip-controller isn’t a good implementation before and it was removed in the master-head.
